uiautomationutilities/perf_tool/MathUtils.cs

```diff
@@ -0,0 +1,111 @@
+            {
+                return 0.0;
+            }
+
+            var stdev = 0.0;
+
+            for (var n = 0; n < values.Count; n++)
+            {
+                stdev += Math.Pow(values[n] - average, 2);
+            }
+
+            stdev = Math.Sqrt(stdev / (values.Count - 1));
+
+            return stdev;
+        }
+        public static double GetPercentile(List<double> values, int percentile)
+        {
+            if (percentile < 0 || percentile > 100)
+            {
+                throw new ArgumentException("Percentile can't be calculated for nonsensical index " + percentile);
+            }
+
+            if (values.Count == 1)
+            {
+                return values[0];
+            }
+
+            values.Sort();
+
+            var n = percentile / 100.0 * (values.Count - 1) + 1.0;
+            double left = 0.0, right = 0.0;
+
+            if (n >= 1)
+            {
+                left = values[(int)Math.Floor(n) - 1];
+                right = values[(int)Math.Floor(n)];
+            }
+            else
+            {
+                left = values[0];
+                right = values[1];
+            }
+
+            if (left == right)
+            {
+                return left;
+            }
+
+            var part = n - Math.Floor(n);
+            return left + part * (right - left);
+        }
+
+        public static List<double> GetMinSdSubset(List<double> values, int factor)
+        {
+            values.Sort();
+            double average;
+            if (values.Count == 1)
+            {
+                average = values[0];
+            }
+
+            if (factor <= 0 || factor > 100)
+            {
+                throw new ArgumentException("Closest can't be calculated for nonsensical index " + factor);
+            }
+
+            var setSize = (int)(0.5 + (float)values.Count * factor / 100);
+            //PerfConsole.LogInfoMessage("For a sample size of {0}, closest set size is {1}", values.Count, setSize);
+
+            if (setSize <= 0)
+            {
+                throw new ArgumentException("Factor - {0} too small for {1} iterations " + factor,
+                    values.Count.ToString());
+            }
+
+            var currLow = 0;
+            var currHigh = setSize - 1;
+            var globalLow = 0;
+            var globalHigh = setSize - 1;
+            double globalSd = int.MaxValue;
+
+            while (currHigh < values.Count)
+            {
+                var currSd = GetStdDeviation(values.GetRange(currLow, setSize));
+                if (currSd <= globalSd)
+                {
+                    globalSd = currSd;
+                    globalLow = currLow;
+                    globalHigh = currHigh;
+                }
+                currHigh++;
+                currLow++;
+            }
+            /*PerfConsole.LogInfoMessage("Closest Subset {0}%/{1} Range is {2} - {3} ", factor, values.Count, globalLow,
+                globalHigh);*/
```

Comment:
remove and check for other instances as well.

Reply:
Taken care of this.